### DISCUSSION TITLE
Bug: ignore everything except specific path

### DISCUSSION THREAD

**Discussion by krokyze** [eslint/eslint#19129] [General] [answered]
### Environment

Node version: v21.7.1
npm version: 10.5.0
Local ESLint version:
Global ESLint version: v9.14.0
Operating System: macOS 15.1

### What parser are you using?

Default (Espree)

### What did you do?

```
import globals from 'globals';
import pluginJs from '@eslint/js';
import pluginVue from 'eslint-plugin-vue';

/** @type {import('eslint').Linter.Config[]} */
export default [
	{
		ignores: [
			'**/*',
			'!assets/frontend/vue/**',
		],
	},
	{
		files: ['**/*.{js,mjs,cjs,vue}'],
	},
	{ languageOptions: { globals: globals.browser } },
	pluginJs.configs.recommended,
	...pluginVue.configs['flat/essential'],
];

```

### What did you expect to happen?

All I'm trying to achieve is to ignore everything except specific path and all it's content recursively. I've tried multiple ways (`"!**/vue/**"`, etc)" but it doesn't seem to work.

### What actually happened?

I'm getting `You are linting ".", but all of the files matching the glob pattern "." are ignored.`

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment, marked as answer]
Similar to `.gitignore`, global ignores apply to directories as well. `**/*` ignores directories `assets/` and `assets/frontend/` so you'll need to unignore them, too:

```js
{
    ignores: [
        '**/*', // ignore everything
        '!assets/', // unignore `assets/`
        '!assets/frontend/', // unignore `assets/frontend/`
        '!assets/frontend/vue/**', // unignore 'assets/frontend/vue/' and its contents
    ],
}
```

**Reply by krokyze** [reply]
Thanks for the quick answer!
